src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -318,12 +350,17 @@ public void BuildCFIMap(ObjectNode node)
 
                 if (_targetPlatform.OperatingSystem == TargetOS.Windows)
                 {
-                    SwitchSection(_nativeObjectWriter, "xdata");
-                    EmitAlignment(4);
-
-                    // TODO: Make sure that the same blobs get folded by emitting them to comdat sections
-
                     string blobSymbolName = "_unwind" + (i++).ToStringInvariant() + _currentNodeName;
+
+                    ObjectNodeSection section = ObjectNodeSection.XDataSection;
+                    if (node.ShouldShareNodeAcrossModules(context) && context.Target.OperatingSystem == TargetOS.Windows)
```

Comment:
I like your idea there but right now that would mean duplicating the ShouldShareAcrossModules code into each Node. I'm answering these questions in the CompilationGroup to centralize the logic that decides when to share. I suppose I could stash some extension methods for TypeDesc / MethodDesc somewhere to answer it.  In any case, I will need to change the details around this in the next phase where I make this feature work more generally.  Perhaps best to leave it as-is and I'll play with the design next PR.